src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -218,8 +225,12 @@ private void init()
         VillagerProfession prof = new VillagerProfession("minecraft:farmer", "minecraft:textures/entity/villager/farmer.png");
         {
             register(prof, 0);
-            (new VillagerCareer(prof, "farmer"    )).init(VanillaTrades.trades[0][0]);
-            (new VillagerCareer(prof, "fisherman" )).init(VanillaTrades.trades[0][1]);
+            // Examples of using addTradingOption in init() if it works - dvatreknerd314
+            addTradingOption("minecraft:farmer", "farmer", VanillaTrades.trades[0][0]);
```

Comment:
The init functions that take the atrocious arrays are intended to be private because this is a horrible thing to do.
Proper method is creating your Profession, then your careers, and then adding trades one by one.